### ISSUE TITLE
Problem with z_score?

### ISSUE THREAD

**Issue by nguyenhongson1902** [facebookresearch/SEAL_OGB#20]
Your graph is trained with batch_size = 32. It means you compressed 32 subgraphs into one large graph, but you only mark z_score as value 1 in 2 positions z[0] = 1, z[1] = 1. However, follow the DRNL, target nodes are labeled as 1, it means with batch_size=32, your z_score is supposed to have 32 values that are equal to 1.

**Comment by nguyenhongson1902** [comment]
But when you use  "_, center_indices = np.unique(batch.cpu().numpy(return_index=True), 
                                x_src = x[center_indices] 
                                x_dst = x[center_indices + 1]"
It did not focus on layers 0 and 1 (where z_scores is labeled), it takes position via "center_indices".I think the z_scores label should be in position "center_indices". Sorry, I don't understand this, you can explain it to me.

**Comment by muhanzhang** [comment]
I am not sure what do you mean by "focus on layers 0 and 1 (where z_scores is labeled)". z_scores are only node labels input to the GNN. They have nothing to do with layers. The center_indices is only used in GCN to retrieve the src and dst node representations from all node embeddings. It also has nothing to do with z_scores.

**Comment by nguyenhongson1902** [comment]
```
######subgraph_1#######
edge_index1 = torch.tensor([[0, 1, 1, 2],
                           [1, 0, 2, 1]], dtype=torch.long)
x1 = torch.tensor([[-1], [0], [1]], dtype=torch.float)
z1 = torch.tensor([1, 1, 5], dtype=torch.float)
data1 = Data(x=x1, edge_index=edge_index1,z=z1)
######subgraph_2#######
edge_index2 = torch.tensor([[0, 1, 2, 2],
                           [1, 2, 1, 0]], dtype=torch.long)
x2 = torch.tensor([[10], [11], [12]], dtype=torch.float)
z2 = torch.tensor([1, 1, 7], dtype=torch.float)
data2 = Data(x=x, edge_index=edge_index,z=z)
```

--> Big Graph from 2 graphs above:
```
edge_index = tensor([[0, 1, 1, 2, 3, 4, 5, 5],[1, 0, 2, 1, 4, 5, 4, 3]])
x = tensor([[-1.],
            [ 0.],
            [ 1.],
            [10.],
            [11.],
            [12.]])
z = tensor([1., 1., 5., 1., 1., 7.]) --> 2 label DRNL --> 4 position values 1
batch = tensor([0, 0, 0, 1, 1, 1]) --> center_indices = [0,3] --> possion of z[0] = 1 and z[1] =1 , z[3] = 1 and z[4] =1 
```
It means you compressed 32 subgraphs into one large graph, so you must have 32 labels DRNL (64 position values 1). And the positions value center_indices, center_indices + 1 are the same as the values of positions having z_score equal to 1.

**Comment by nguyenhongson1902** [comment]
Now I understand it.Thank you for the time!
